db/migrate/20210411213625_change_devices_app_bundle_and_add_reference.rb

```diff
@@ -0,0 +1,14 @@
+class ChangeDevicesAppBundleAndAddReference < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  def change
+    add_reference :devices, :app_integration, index: {algorithm: :concurrently}
+
+    # Allow app_bundle null values and remove composite index
+    change_column_null :devices, :app_bundle, true
```

Comment:
I'm not sure these two should run outside a transaction, they're not indexes. The first actually adds a foreing key constraint and the second runs an alter table on a column. As much as devices is a small table, we shouldn't make a habit of running data manipulation statements outside of DB transactions. If anything happens we'll leave the DB in a state that might require human intervention. 

I suggest to move these in a regular migration

Reply:
@rhymes this took a bit of back and forth (me vs `strong_migrations` in my local env) to work okay.

> The first actually adds a foreing key constraint and the second runs an alter table on a column.

> I suggest to move these in a regular migration

I'm not sure I follow your suggestion here. `ChangeDevicesAppBundleAndAddReference` adds a reference (**without** adding the Foreign Key constraint) and then `AddDevicesForeignKeyToAppIntegrations` (next migration) actually adds the Foreign Key Constraint (the validation comes in a separate migration too).

These were all `strong_migrations` suggestions (more like it failed and forced me to separate things this way).

Would your suggestion be to wrap the FK Constraint creation or the `change_column_null` (or both) in a Transaction? Also make all the FK Constraint changes in a single migration wrapped by a `safety_assured` block to avoid `strong_migrations` raising an error?

Reply:
@fdoxyz oh gosh, you're right! `add_reference` defaults to `foreign_key: false`, make sense.

Maybe we should move `change_column_null :devices, :app_bundle, true` in its own migration, `strong_migrations` correctly complains with setting `NOT NULL`, not the other way around. The rest can stay in this migration file.

Would that work?

Reply:
Makes sense to me! I'll give it a go when I refactor this PR again soon 💯